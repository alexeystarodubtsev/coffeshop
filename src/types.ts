export interface Goods {
    imageUrl: string;
    name: string;
    price: number;
    info: string;
    tax: number;
}

export interface GoodsInBasket {
    name: string;
    qty: number;
}

export type OrderStatus = 'MAKING' | 'PREPARING' | 'DONE';

export interface Notification {
    title: string;
    date: Date;
}
