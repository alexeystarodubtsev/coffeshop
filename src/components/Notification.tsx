import React, { useState, useEffect } from 'react';
import Button from '@material-ui/core/Button';
import ClearIcon from '@material-ui/icons/Clear';
import { Notification as NotificationType } from '../types';
import Zoom from '@material-ui/core/Zoom';

interface Props {
    onClose: () => void;
    notification: NotificationType;
}

const CLASS = 'notifications';
const timeoutSec = 1000;
export const Notification: React.FC<Props> = ({ onClose, notification: { title } }) => {
    const [open, setState] = useState(false);
    useEffect(() => {
        setState(true);
    }, []);
    const handleClose = () => {
        setState(false);
        setTimeout(() => {
            onClose();
        }, timeoutSec);
    };

    return (
        <Zoom in={open} timeout={timeoutSec}>
            <div className={`${CLASS}__item`}>
                {title}
                <Button className={`${CLASS}__item-close-btn`} color="primary" size="small" onClick={handleClose}>
                    <ClearIcon />
                </Button>
            </div>
        </Zoom>
    );
};
