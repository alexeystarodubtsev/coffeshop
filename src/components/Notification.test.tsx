import { render } from '@testing-library/react';
import { Notification } from './Notification';
import React from 'react';
import { Notification as NotificationType } from '../types';
import '@testing-library/jest-dom/extend-expect';

describe('Notifications', () => {
    const notification: NotificationType = { title: 'test notification', date: new Date() };
    it('Appearing notification', () => {
        const { container, getByText } = render(<Notification notification={notification} onClose={() => undefined} />);
        expect(container.firstChild).toMatchSnapshot();
        expect(container.firstChild).toBeDefined();
        expect(getByText('test notification')).toBeInTheDocument();
    });
});
