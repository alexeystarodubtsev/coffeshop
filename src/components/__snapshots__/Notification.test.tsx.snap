// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Appearing notification 1`] = `
<div
  class="notifications__item"
  style="transform: none; webkit-transition: transform 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
>
  test notification
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text notifications__item-close-btn MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
